src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -662,6 +662,20 @@ bool IdealLoopTree::is_range_check_if(IfNode *iff, PhaseIdealLoop *phase, Invari
   if (offset && !invar.is_invariant(offset)) { // offset must be invariant
     return false;
   }
+#ifndef PRODUCT
```

Comment:
Should be `#ifdef ASSERT` for debug build.

Reply:
On other hand, consider do this check in PRODUCT too to bailout (return `false`).